// @ark-ui/vue/color-picker/use-color-picker.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const colorPicker = require('@zag-js/color-picker');
const vue$1 = require('@zag-js/vue');
const vue = require('vue');
const environmentContext = require('../environment/environment-context.cjs');
const utils = require('../utils.cjs');

function _interopNamespaceDefault(e) {
  const n = Object.create(null, { [Symbol.toStringTag]: { value: 'Module' } });
  if (e) {
    for (const k in e) {
      if (k !== 'default') {
        const d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: () => e[k]
        });
      }
    }
  }
  n.default = e;
  return Object.freeze(n);
}

const colorPicker__namespace = /*#__PURE__*/_interopNamespaceDefault(colorPicker);

const useColorPicker = (props, emit) => {
  const getRootNode = environmentContext.useEnvironmentContext();
  const context = vue.computed(() => {
    const { modelValue, ...rest } = props;
    return {
      ...rest,
      value: modelValue ? colorPicker__namespace.parse(modelValue) : void 0
    };
  });
  const [state, send] = vue$1.useMachine(
    colorPicker__namespace.machine({
      ...context.value,
      id: context.value.id ?? utils.useId().value,
      getRootNode,
      onFormatChange(details) {
        emit("format-change", details);
      },
      onOpenChange(details) {
        emit("open-change", details);
      },
      onValueChange(details) {
        emit("value-change", details);
        emit("update:modelValue", details.valueAsString);
      },
      onValueChangeEnd(details) {
        emit("value-change-end", details);
      }
    }),
    { context }
  );
  return vue.computed(() => colorPicker__namespace.connect(state.value, send, vue$1.normalizeProps));
};

exports.useColorPicker = useColorPicker;
