// @ark-ui/vue/tags-input/tags-input.props.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const utils = require('../utils.cjs');

const props = {
  addOnPaste: {
    type: Boolean,
    default: void 0
  },
  allowEditTag: {
    type: Boolean,
    default: void 0
  },
  allowOverflow: {
    type: Boolean,
    default: void 0
  },
  autoFocus: {
    type: Boolean,
    default: void 0
  },
  blurBehavior: {
    type: String
  },
  delimiter: {
    type: String
  },
  dir: {
    type: String
  },
  disabled: {
    type: Boolean,
    default: void 0
  },
  form: {
    type: String
  },
  getRootNode: {
    type: Function
  },
  id: {
    type: String
  },
  ids: {
    type: Object
  },
  inputValue: {
    type: String
  },
  invalid: {
    type: Boolean,
    default: void 0
  },
  max: {
    type: Number
  },
  maxLength: {
    type: Number
  },
  name: {
    type: String
  },
  readOnly: {
    type: Boolean,
    default: void 0
  },
  translations: {
    type: Object
  },
  validate: {
    type: Function
  },
  modelValue: {
    type: Array
  }
};
const emits = utils.declareEmits([
  "highlight-change",
  "value-change",
  "value-invalid",
  "update:modelValue"
]);

exports.emits = emits;
exports.props = props;
