// @ark-ui/vue/hover-card/hover-card-root.mjs
import { defineComponent, computed } from 'vue';
import { PresenceProvider } from '../presence/presence-context.mjs';
import { usePresence } from '../presence/use-presence.mjs';
import { props as props$1, emits as emits$1 } from '../presence/presence.props.mjs';
import { HoverCardProvider } from './hover-card-context.mjs';
import { props, emits } from './hover-card.props.mjs';
import { useHoverCard } from './use-hover-card.mjs';

const HoverCardRoot = /* @__PURE__ */ defineComponent((props, {
  slots,
  emit
}) => {
  const api = useHoverCard(props, emit);
  const isOpen = computed(() => api.value.isOpen);
  const presenceProps = computed(() => ({
    present: props.present || isOpen.value,
    lazyMount: props.lazyMount,
    unmountOnExit: props.unmountOnExit
  }));
  const presenceApi = usePresence(presenceProps, emit);
  HoverCardProvider(api);
  PresenceProvider(presenceApi);
  return () => slots.default?.(api.value);
}, {
  name: 'HoverCardRoot',
  props: {
    ...props,
    ...props$1
  },
  emits: [...emits, ...emits$1]
});

export { HoverCardRoot };
