// @ark-ui/vue/hover-card/hover-card-root.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const vue = require('vue');
const presenceContext = require('../presence/presence-context.cjs');
const usePresence = require('../presence/use-presence.cjs');
const presence_props = require('../presence/presence.props.cjs');
const hoverCardContext = require('./hover-card-context.cjs');
const hoverCard_props = require('./hover-card.props.cjs');
const useHoverCard = require('./use-hover-card.cjs');

const HoverCardRoot = /* @__PURE__ */ vue.defineComponent((props, {
  slots,
  emit
}) => {
  const api = useHoverCard.useHoverCard(props, emit);
  const isOpen = vue.computed(() => api.value.isOpen);
  const presenceProps = vue.computed(() => ({
    present: props.present || isOpen.value,
    lazyMount: props.lazyMount,
    unmountOnExit: props.unmountOnExit
  }));
  const presenceApi = usePresence.usePresence(presenceProps, emit);
  hoverCardContext.HoverCardProvider(api);
  presenceContext.PresenceProvider(presenceApi);
  return () => slots.default?.(api.value);
}, {
  name: 'HoverCardRoot',
  props: {
    ...hoverCard_props.props,
    ...presence_props.props
  },
  emits: [...hoverCard_props.emits, ...presence_props.emits]
});

exports.HoverCardRoot = HoverCardRoot;
