// @ark-ui/vue/number-input/number-input.props.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const utils = require('../utils.cjs');

const props = {
  allowMouseWheel: {
    type: Boolean,
    default: void 0
  },
  allowOverflow: {
    type: Boolean,
    default: void 0
  },
  clampValueOnBlur: {
    type: Boolean,
    default: void 0
  },
  dir: {
    type: String
  },
  disabled: {
    type: Boolean,
    default: void 0
  },
  focusInputOnChange: {
    type: Boolean,
    default: void 0
  },
  form: {
    type: String
  },
  formatOptions: {
    type: Object
  },
  getRootNode: {
    type: Function
  },
  id: {
    type: String
  },
  ids: {
    type: Object
  },
  inputMode: {
    type: String
  },
  invalid: {
    type: Boolean,
    default: void 0
  },
  locale: {
    type: String
  },
  max: {
    type: Number
  },
  min: {
    type: Number
  },
  name: {
    type: String
  },
  pattern: {
    type: String
  },
  readOnly: {
    type: Boolean,
    default: void 0
  },
  spinOnPress: {
    type: Boolean,
    default: void 0
  },
  step: {
    type: Number
  },
  translations: {
    type: Object
  },
  modelValue: {
    type: String
  }
};
const emits = utils.declareEmits([
  "focus-change",
  "value-change",
  "value-invalid",
  "update:modelValue"
]);

exports.emits = emits;
exports.props = props;
