// @ark-ui/vue/slider/slider.props.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const utils = require('../utils.cjs');

const props = {
  dir: {
    type: String
  },
  disabled: {
    type: Boolean,
    default: void 0
  },
  form: {
    type: String
  },
  getAriaValueText: {
    type: Function
  },
  getRootNode: {
    type: Function
  },
  id: {
    type: String
  },
  ids: {
    type: Object
  },
  invalid: {
    type: Boolean,
    default: void 0
  },
  max: {
    type: Number
  },
  min: {
    type: Number
  },
  minStepsBetweenThumbs: {
    type: Number
  },
  name: {
    type: String
  },
  orientation: {
    type: String
  },
  origin: {
    type: String
  },
  readOnly: {
    type: Boolean,
    default: void 0
  },
  step: {
    type: Number
  },
  thumbAlignment: {
    type: String
  },
  thumbSize: {
    type: Object
  },
  modelValue: {
    type: Array
  }
};
const emits = utils.declareEmits([
  "focus-change",
  "value-change-end",
  "value-change",
  "update:modelValue"
]);

exports.emits = emits;
exports.props = props;
