// @ark-ui/vue/tabs/tab-content.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const vue = require('vue');
const factory = require('../factory.cjs');
const presenceContext = require('../presence/presence-context.cjs');
const presencePropsContext = require('../presence/presence-props-context.cjs');
const usePresence = require('../presence/use-presence.cjs');
const presence_props = require('../presence/presence.props.cjs');
const tabsContext = require('./tabs-context.cjs');

const TabContent = /* @__PURE__ */ vue.defineComponent((props, {
  slots,
  attrs,
  emit
}) => {
  const api = tabsContext.useTabsContext();
  const presenceProps = presencePropsContext.usePresencePropsContext();
  const usePresenceProps = vue.computed(() => ({
    ...presenceProps.value,
    present: api.value.value === props.value
  }));
  const presenceApi = usePresence.usePresence(usePresenceProps, emit);
  presenceContext.PresenceProvider(presenceApi);
  return () => vue.createVNode(vue.Fragment, null, [presenceApi.value.isUnmounted ? null : vue.createVNode(factory.ark.div, vue.mergeProps(api.value.getContentProps(props), attrs), {
    default: () => [slots.default?.()]
  })]);
}, {
  name: 'TabContent',
  props: {
    value: {
      type: String,
      required: true
    }
  },
  emits: presence_props.emits
});

exports.TabContent = TabContent;
