// @ark-ui/vue/tabs/tabs-root.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const vue = require('vue');
const factory = require('../factory.cjs');
const presencePropsContext = require('../presence/presence-props-context.cjs');
const presence_props = require('../presence/presence.props.cjs');
const tabsContext = require('./tabs-context.cjs');
const tabs_props = require('./tabs.props.cjs');
const useTabs = require('./use-tabs.cjs');

const TabsRoot = /* @__PURE__ */ vue.defineComponent((props, {
  slots,
  attrs,
  emit
}) => {
  const api = useTabs.useTabs(props, emit);
  const presenceProps = vue.computed(() => ({
    present: props.present,
    lazyMount: props.lazyMount,
    unmountOnExit: props.unmountOnExit
  }));
  tabsContext.TabsProvider(api);
  presencePropsContext.PresencePropsProvider(presenceProps);
  return () => vue.createVNode(factory.ark.div, vue.mergeProps(api.value.rootProps, attrs), {
    default: () => [slots.default?.(api.value)]
  });
}, {
  name: 'TabsRoot',
  props: {
    ...tabs_props.props,
    ...presence_props.props
  },
  emits: [...tabs_props.emits, ...presence_props.emits]
});

exports.TabsRoot = TabsRoot;
