// @ark-ui/vue/popover/popover-root.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const vue = require('vue');
const presenceContext = require('../presence/presence-context.cjs');
const usePresence = require('../presence/use-presence.cjs');
const presence_props = require('../presence/presence.props.cjs');
const popoverContext = require('./popover-context.cjs');
const popover_props = require('./popover.props.cjs');
const usePopover = require('./use-popover.cjs');

const PopoverRoot = /* @__PURE__ */ vue.defineComponent((props, {
  slots,
  emit
}) => {
  const api = usePopover.usePopover(props, emit);
  const isOpen = vue.computed(() => api.value.isOpen);
  const presenceProps = vue.computed(() => ({
    present: props.present || isOpen.value,
    lazyMount: props.lazyMount,
    unmountOnExit: props.unmountOnExit
  }));
  const presenceApi = usePresence.usePresence(presenceProps, emit);
  popoverContext.PopoverProvider(api);
  presenceContext.PresenceProvider(presenceApi);
  return () => slots.default?.(api.value);
}, {
  name: 'PopoverRoot',
  props: {
    ...popover_props.props,
    ...presence_props.props
  },
  emits: [...popover_props.emits, ...presence_props.emits]
});

exports.PopoverRoot = PopoverRoot;
