// @ark-ui/vue/popover/popover-root.mjs
import { defineComponent, computed } from 'vue';
import { PresenceProvider } from '../presence/presence-context.mjs';
import { usePresence } from '../presence/use-presence.mjs';
import { props as props$1, emits as emits$1 } from '../presence/presence.props.mjs';
import { PopoverProvider } from './popover-context.mjs';
import { props, emits } from './popover.props.mjs';
import { usePopover } from './use-popover.mjs';

const PopoverRoot = /* @__PURE__ */ defineComponent((props, {
  slots,
  emit
}) => {
  const api = usePopover(props, emit);
  const isOpen = computed(() => api.value.isOpen);
  const presenceProps = computed(() => ({
    present: props.present || isOpen.value,
    lazyMount: props.lazyMount,
    unmountOnExit: props.unmountOnExit
  }));
  const presenceApi = usePresence(presenceProps, emit);
  PopoverProvider(api);
  PresenceProvider(presenceApi);
  return () => slots.default?.(api.value);
}, {
  name: 'PopoverRoot',
  props: {
    ...props,
    ...props$1
  },
  emits: [...emits, ...emits$1]
});

export { PopoverRoot };
