// @ark-ui/vue/popover/use-popover.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const popover = require('@zag-js/popover');
const vue$1 = require('@zag-js/vue');
const vue = require('vue');
const environmentContext = require('../environment/environment-context.cjs');
const utils = require('../utils.cjs');

function _interopNamespaceDefault(e) {
  const n = Object.create(null, { [Symbol.toStringTag]: { value: 'Module' } });
  if (e) {
    for (const k in e) {
      if (k !== 'default') {
        const d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: () => e[k]
        });
      }
    }
  }
  n.default = e;
  return Object.freeze(n);
}

const popover__namespace = /*#__PURE__*/_interopNamespaceDefault(popover);

const usePopover = (props, emit) => {
  const getRootNode = environmentContext.useEnvironmentContext();
  const context = vue.ref(props);
  const [state, send] = vue$1.useMachine(
    popover__namespace.machine({
      ...context.value,
      id: context.value.id || utils.useId().value,
      open: props.open ?? props.defaultOpen,
      getRootNode,
      onOpenChange: (details) => {
        emit("open-change", details);
        emit("update:open", details.open);
      },
      onEscapeKeyDown: (details) => {
        emit("escape-key-down", details);
      },
      onFocusOutside: (details) => {
        emit("focus-outside", details);
      },
      onInteractOutside: (details) => {
        emit("interact-outside", details);
      },
      onPointerDownOutside: (details) => {
        emit("pointer-down-outside", details);
      }
    }),
    { context }
  );
  return vue.computed(() => popover__namespace.connect(state.value, send, vue$1.normalizeProps));
};

exports.usePopover = usePopover;
