// @ark-ui/vue/editable/editable.props.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const utils = require('../utils.cjs');

const props = {
  activationMode: {
    type: String
  },
  autoResize: {
    type: Boolean,
    default: void 0
  },
  dir: {
    type: String
  },
  disabled: {
    type: Boolean,
    default: void 0
  },
  finalFocusEl: {
    type: Function
  },
  form: {
    type: String
  },
  getRootNode: {
    type: Function
  },
  id: {
    type: String
  },
  ids: {
    type: Object
  },
  invalid: {
    type: Boolean,
    default: void 0
  },
  maxLength: {
    type: Number
  },
  name: {
    type: String
  },
  placeholder: {
    type: String
  },
  readOnly: {
    type: Boolean,
    default: void 0
  },
  selectOnFocus: {
    type: Boolean,
    default: void 0
  },
  startWithEditView: {
    type: Boolean,
    default: void 0
  },
  submitMode: {
    type: String
  },
  translations: {
    type: Object
  },
  modelValue: {
    type: String
  }
};
const emits = utils.declareEmits([
  "edit",
  "value-change",
  "value-commit",
  "value-revert",
  "update:modelValue"
]);

exports.emits = emits;
exports.props = props;
