// @ark-ui/vue/editable/use-editable.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const editable = require('@zag-js/editable');
const vue$1 = require('@zag-js/vue');
const vue = require('vue');
const environmentContext = require('../environment/environment-context.cjs');
const utils = require('../utils.cjs');

function _interopNamespaceDefault(e) {
  const n = Object.create(null, { [Symbol.toStringTag]: { value: 'Module' } });
  if (e) {
    for (const k in e) {
      if (k !== 'default') {
        const d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: () => e[k]
        });
      }
    }
  }
  n.default = e;
  return Object.freeze(n);
}

const editable__namespace = /*#__PURE__*/_interopNamespaceDefault(editable);

const useEditable = (props, emit) => {
  const getRootNode = environmentContext.useEnvironmentContext();
  const context = vue.computed(() => {
    const { modelValue, ...rest } = props;
    return {
      ...rest,
      value: modelValue
    };
  });
  const [state, send] = vue$1.useMachine(
    editable__namespace.machine({
      ...context.value,
      id: context.value.id ?? utils.useId().value,
      getRootNode,
      onValueChange(details) {
        emit("value-change", details);
        emit("update:modelValue", details.value);
      },
      onValueCommit(details) {
        emit("value-commit", details);
      },
      onValueRevert(details) {
        emit("value-revert", details);
      },
      onEdit() {
        emit("edit");
      }
    }),
    { context }
  );
  return vue.computed(() => editable__namespace.connect(state.value, send, vue$1.normalizeProps));
};

exports.useEditable = useEditable;
