// @ark-ui/vue/editable/use-editable.mjs
import * as editable from '@zag-js/editable';
import { useMachine, normalizeProps } from '@zag-js/vue';
import { computed } from 'vue';
import { useEnvironmentContext } from '../environment/environment-context.mjs';
import { useId } from '../utils.mjs';

const useEditable = (props, emit) => {
  const getRootNode = useEnvironmentContext();
  const context = computed(() => {
    const { modelValue, ...rest } = props;
    return {
      ...rest,
      value: modelValue
    };
  });
  const [state, send] = useMachine(
    editable.machine({
      ...context.value,
      id: context.value.id ?? useId().value,
      getRootNode,
      onValueChange(details) {
        emit("value-change", details);
        emit("update:modelValue", details.value);
      },
      onValueCommit(details) {
        emit("value-commit", details);
      },
      onValueRevert(details) {
        emit("value-revert", details);
      },
      onEdit() {
        emit("edit");
      }
    }),
    { context }
  );
  return computed(() => editable.connect(state.value, send, normalizeProps));
};

export { useEditable };
