// @ark-ui/vue/tooltip/tooltip-root.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const vue = require('vue');
const presenceContext = require('../presence/presence-context.cjs');
const usePresence = require('../presence/use-presence.cjs');
const presence_props = require('../presence/presence.props.cjs');
const tooltipContext = require('./tooltip-context.cjs');
const tooltip_props = require('./tooltip.props.cjs');
const useTooltip = require('./use-tooltip.cjs');

const TooltipRoot = /* @__PURE__ */ vue.defineComponent((props, {
  slots,
  emit
}) => {
  const api = useTooltip.useTooltip(props, emit);
  const isOpen = vue.computed(() => api.value.isOpen);
  const presenceProps = vue.computed(() => ({
    present: props.present || isOpen.value,
    lazyMount: props.lazyMount,
    unmountOnExit: props.unmountOnExit
  }));
  const presenceApi = usePresence.usePresence(presenceProps, emit);
  tooltipContext.TooltipProvider(api);
  presenceContext.PresenceProvider(presenceApi);
  return () => slots.default?.(api.value);
}, {
  name: 'TooltipRoot',
  props: {
    ...tooltip_props.props,
    ...presence_props.props
  },
  emits: [...tooltip_props.emits, ...presence_props.emits]
});

exports.TooltipRoot = TooltipRoot;
