// @ark-ui/vue/tooltip/tooltip-root.mjs
import { defineComponent, computed } from 'vue';
import { PresenceProvider } from '../presence/presence-context.mjs';
import { usePresence } from '../presence/use-presence.mjs';
import { props as props$1, emits as emits$1 } from '../presence/presence.props.mjs';
import { TooltipProvider } from './tooltip-context.mjs';
import { props, emits } from './tooltip.props.mjs';
import { useTooltip } from './use-tooltip.mjs';

const TooltipRoot = /* @__PURE__ */ defineComponent((props, {
  slots,
  emit
}) => {
  const api = useTooltip(props, emit);
  const isOpen = computed(() => api.value.isOpen);
  const presenceProps = computed(() => ({
    present: props.present || isOpen.value,
    lazyMount: props.lazyMount,
    unmountOnExit: props.unmountOnExit
  }));
  const presenceApi = usePresence(presenceProps, emit);
  TooltipProvider(api);
  PresenceProvider(presenceApi);
  return () => slots.default?.(api.value);
}, {
  name: 'TooltipRoot',
  props: {
    ...props,
    ...props$1
  },
  emits: [...emits, ...emits$1]
});

export { TooltipRoot };
