// @ark-ui/vue/menu/menu-option-item.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const vue = require('vue');
const factory = require('../factory.cjs');
const menuContext = require('./menu-context.cjs');

const MenuOptionItem = /* @__PURE__ */ vue.defineComponent((props, {
  slots,
  attrs,
  emit
}) => {
  const menuOptionItemProps = vue.computed(() => ({
    ...props,
    onCheckedChange(checked) {
      emit('checked-change', checked);
    }
  }));
  const api = menuContext.useMenuContext();
  return () => vue.createVNode(factory.ark.div, vue.mergeProps(api.value.getOptionItemProps(menuOptionItemProps.value), attrs), {
    default: () => [slots.default?.(api.value.getOptionItemState(menuOptionItemProps.value))]
  });
}, {
  name: 'MenuOptionItem',
  props: {
    id: {
      type: String
    },
    disabled: {
      type: Boolean,
      default: undefined
    },
    valueText: {
      type: String,
      default: undefined
    },
    closeOnSelect: {
      type: Boolean,
      default: undefined
    },
    name: {
      type: String,
      required: true
    },
    type: {
      type: String,
      required: true
    },
    value: {
      type: String,
      required: true
    }
  },
  emits: ['checked-change']
});

exports.MenuOptionItem = MenuOptionItem;
