// @ark-ui/vue/select/select.props.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const utils = require('../utils.cjs');

const props = {
  closeOnSelect: {
    type: Boolean,
    default: void 0
  },
  dir: {
    type: String
  },
  disabled: {
    type: Boolean,
    default: void 0
  },
  form: {
    type: String
  },
  getRootNode: {
    type: Function
  },
  highlightedValue: {
    type: String
  },
  id: {
    type: String
  },
  ids: {
    type: Object
  },
  invalid: {
    type: Boolean,
    default: void 0
  },
  loop: {
    type: Boolean,
    default: void 0
  },
  multiple: {
    type: Boolean,
    default: void 0
  },
  name: {
    type: String
  },
  open: {
    type: Boolean,
    default: void 0
  },
  positioning: {
    type: Object
  },
  readOnly: {
    type: Boolean,
    default: void 0
  },
  selectOnBlur: {
    type: Boolean,
    default: void 0
  },
  modelValue: {
    type: Array
  }
};
const emits = utils.declareEmits([
  "highlight-change",
  "open-change",
  "value-change",
  "update:modelValue"
]);

exports.emits = emits;
exports.props = props;
