// @ark-ui/vue/select/select-root.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const vue = require('vue');
const factory = require('../factory.cjs');
const presenceContext = require('../presence/presence-context.cjs');
const usePresence = require('../presence/use-presence.cjs');
const presence_props = require('../presence/presence.props.cjs');
const selectContext = require('./select-context.cjs');
const select_props = require('./select.props.cjs');
const useSelect = require('./use-select.cjs');

// TOOD: #2011 this is a bad workaround but should work for now
// function signature doesn't really support more complicated generics
const SelectRoot = /* @__PURE__ */ vue.defineComponent((props, {
  slots,
  attrs,
  emit
}) => {
  const api = useSelect.useSelect(props, emit);
  const presenceProps = vue.computed(() => ({
    present: props.present || api.value.isOpen,
    lazyMount: props.lazyMount,
    unmountOnExit: props.unmountOnExit
  }));
  const presenceApi = usePresence.usePresence(presenceProps, emit);
  selectContext.SelectProvider(api);
  presenceContext.PresenceProvider(presenceApi);
  return () => vue.createVNode(factory.ark.div, vue.mergeProps(api.value.rootProps, attrs), {
    default: () => [slots.default?.(api.value)]
  });
}, {
  name: 'SelectRoot',
  props: {
    ...select_props.props,
    ...presence_props.props,
    items: {
      type: Array,
      required: true
    },
    itemToString: {
      type: Function
    },
    itemToValue: {
      type: Function
    },
    isItemDisabled: {
      type: Function
    }
  },
  emits: [...select_props.emits, ...presence_props.emits]
});

exports.SelectRoot = SelectRoot;
