// @ark-ui/vue/select/select-root.mjs
import { defineComponent, computed, createVNode, mergeProps } from 'vue';
import { ark } from '../factory.mjs';
import { PresenceProvider } from '../presence/presence-context.mjs';
import { usePresence } from '../presence/use-presence.mjs';
import { props as props$1, emits as emits$1 } from '../presence/presence.props.mjs';
import { SelectProvider } from './select-context.mjs';
import { props, emits } from './select.props.mjs';
import { useSelect } from './use-select.mjs';

// TOOD: #2011 this is a bad workaround but should work for now
// function signature doesn't really support more complicated generics
const SelectRoot = /* @__PURE__ */ defineComponent((props, {
  slots,
  attrs,
  emit
}) => {
  const api = useSelect(props, emit);
  const presenceProps = computed(() => ({
    present: props.present || api.value.isOpen,
    lazyMount: props.lazyMount,
    unmountOnExit: props.unmountOnExit
  }));
  const presenceApi = usePresence(presenceProps, emit);
  SelectProvider(api);
  PresenceProvider(presenceApi);
  return () => createVNode(ark.div, mergeProps(api.value.rootProps, attrs), {
    default: () => [slots.default?.(api.value)]
  });
}, {
  name: 'SelectRoot',
  props: {
    ...props,
    ...props$1,
    items: {
      type: Array,
      required: true
    },
    itemToString: {
      type: Function
    },
    itemToValue: {
      type: Function
    },
    isItemDisabled: {
      type: Function
    }
  },
  emits: [...emits, ...emits$1]
});

export { SelectRoot };
