// @ark-ui/vue/select/use-select.mjs
import * as select from '@zag-js/select';
import { useMachine, normalizeProps } from '@zag-js/vue';
import { computed } from 'vue';
import { useEnvironmentContext } from '../environment/environment-context.mjs';
import { useId } from '../utils.mjs';

const useSelect = (props, emit) => {
  const context = computed(() => {
    const { items, itemToString, itemToValue, isItemDisabled, modelValue, ...rest } = props;
    return {
      ...rest,
      collection: select.collection({ items, itemToString, itemToValue, isItemDisabled }),
      value: modelValue
    };
  });
  const getRootNode = useEnvironmentContext();
  const [state, send] = useMachine(
    select.machine({
      ...context.value,
      id: context.value.id ?? useId().value,
      getRootNode,
      onValueChange: (details) => {
        emit("value-change", details);
        emit("update:modelValue", details.value);
      },
      onHighlightChange: (details) => {
        emit("highlight-change", details);
      },
      onOpenChange: (details) => {
        emit("open-change", details);
      }
    }),
    { context }
  );
  return computed(() => select.connect(state.value, send, normalizeProps));
};

export { useSelect };
