// @ark-ui/vue/combobox/use-combobox.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const combobox = require('@zag-js/combobox');
const vue$1 = require('@zag-js/vue');
const vue = require('vue');
const environmentContext = require('../environment/environment-context.cjs');
const utils = require('../utils.cjs');

function _interopNamespaceDefault(e) {
  const n = Object.create(null, { [Symbol.toStringTag]: { value: 'Module' } });
  if (e) {
    for (const k in e) {
      if (k !== 'default') {
        const d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: () => e[k]
        });
      }
    }
  }
  n.default = e;
  return Object.freeze(n);
}

const combobox__namespace = /*#__PURE__*/_interopNamespaceDefault(combobox);

const useCombobox = (props, emit) => {
  const getRootNode = environmentContext.useEnvironmentContext();
  const context = vue.computed(() => {
    const { items, itemToString, itemToValue, isItemDisabled, modelValue, ...rest } = props;
    return {
      ...rest,
      collection: combobox__namespace.collection({ items, itemToString, itemToValue, isItemDisabled }),
      value: modelValue
    };
  });
  const [state, send] = vue$1.useMachine(
    combobox__namespace.machine({
      ...context.value,
      id: context.value.id ?? utils.useId().value,
      getRootNode,
      onHighlightChange: (details) => {
        emit("highlight-change", details);
      },
      onInputValueChange: (details) => {
        emit("input-value-change", details);
      },
      onOpenChange: (details) => {
        emit("open-change", details);
      },
      onValueChange: (details) => {
        emit("value-change", details);
        emit("update:modelValue", details.value);
      }
    }),
    { context }
  );
  return vue.computed(() => combobox__namespace.connect(state.value, send, vue$1.normalizeProps));
};

exports.useCombobox = useCombobox;
