// @ark-ui/vue/accordion/accordion-root.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const vue = require('vue');
const factory = require('../factory.cjs');
const presencePropsContext = require('../presence/presence-props-context.cjs');
const presence_props = require('../presence/presence.props.cjs');
const accordionContext = require('./accordion-context.cjs');
const accordion_props = require('./accordion.props.cjs');
const useAccordion = require('./use-accordion.cjs');

const AccordionRoot = /* @__PURE__ */ vue.defineComponent((props, {
  slots,
  attrs,
  emit
}) => {
  const api = useAccordion.useAccordion(props, emit);
  const presenceProps = vue.computed(() => ({
    present: props.present,
    lazyMount: props.lazyMount,
    unmountOnExit: props.unmountOnExit
  }));
  accordionContext.AccordionProvider(api);
  presencePropsContext.PresencePropsProvider(presenceProps);
  return () => vue.createVNode(factory.ark.div, vue.mergeProps(api.value.rootProps, attrs), {
    default: () => [slots.default?.(api.value)]
  });
}, {
  name: 'AccordionRoot',
  props: {
    ...accordion_props.props,
    ...presence_props.props
  },
  emits: [...accordion_props.emits, ...presence_props.emits]
});

exports.AccordionRoot = AccordionRoot;
