// @ark-ui/vue/accordion/accordion-root.mjs
import { defineComponent, computed, createVNode, mergeProps } from 'vue';
import { ark } from '../factory.mjs';
import { PresencePropsProvider } from '../presence/presence-props-context.mjs';
import { props as props$1, emits as emits$1 } from '../presence/presence.props.mjs';
import { AccordionProvider } from './accordion-context.mjs';
import { props, emits } from './accordion.props.mjs';
import { useAccordion } from './use-accordion.mjs';

const AccordionRoot = /* @__PURE__ */ defineComponent((props, {
  slots,
  attrs,
  emit
}) => {
  const api = useAccordion(props, emit);
  const presenceProps = computed(() => ({
    present: props.present,
    lazyMount: props.lazyMount,
    unmountOnExit: props.unmountOnExit
  }));
  AccordionProvider(api);
  PresencePropsProvider(presenceProps);
  return () => createVNode(ark.div, mergeProps(api.value.rootProps, attrs), {
    default: () => [slots.default?.(api.value)]
  });
}, {
  name: 'AccordionRoot',
  props: {
    ...props,
    ...props$1
  },
  emits: [...emits, ...emits$1]
});

export { AccordionRoot };
