// @ark-ui/vue/accordion/index.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const accordionContext = require('./accordion-context.cjs');
const accordionItem = require('./accordion-item.cjs');
const accordionItemContent = require('./accordion-item-content.cjs');
const accordionItemContext = require('./accordion-item-context.cjs');
const accordionItemIndicator = require('./accordion-item-indicator.cjs');
const accordionItemTrigger = require('./accordion-item-trigger.cjs');
const accordionRoot = require('./accordion-root.cjs');
const accordion = require('./accordion.cjs');



exports.useAccordionContext = accordionContext.useAccordionContext;
exports.AccordionItem = accordionItem.AccordionItem;
exports.AccordionItemContent = accordionItemContent.AccordionItemContent;
exports.useAccordionItemContext = accordionItemContext.useAccordionItemContext;
exports.AccordionItemIndicator = accordionItemIndicator.AccordionItemIndicator;
exports.AccordionItemTrigger = accordionItemTrigger.AccordionItemTrigger;
exports.AccordionRoot = accordionRoot.AccordionRoot;
exports.Accordion = accordion;
