// @ark-ui/vue/pin-input/pin-input.props.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const utils = require('../utils.cjs');

const props = {
  autoFocus: {
    type: Boolean,
    default: void 0
  },
  blurOnComplete: {
    type: Boolean,
    default: void 0
  },
  dir: {
    type: String
  },
  disabled: {
    type: Boolean,
    default: void 0
  },
  form: {
    type: String
  },
  getRootNode: {
    type: Function
  },
  id: {
    type: String
  },
  ids: {
    type: Object
  },
  invalid: {
    type: Boolean,
    default: void 0
  },
  mask: {
    type: Boolean,
    default: void 0
  },
  name: {
    type: String
  },
  otp: {
    type: Boolean,
    default: void 0
  },
  pattern: {
    type: String
  },
  placeholder: {
    type: String
  },
  selectOnFocus: {
    type: Boolean,
    default: void 0
  },
  translations: {
    type: Object
  },
  type: {
    type: String
  },
  modelValue: {
    type: Array
  }
};
const emits = utils.declareEmits([
  "value-change",
  "value-complete",
  "value-invalid",
  "update:modelValue"
]);

exports.emits = emits;
exports.props = props;
