// @ark-ui/vue/date-picker/date-picker-root.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const vue = require('vue');
const factory = require('../factory.cjs');
const presenceContext = require('../presence/presence-context.cjs');
const usePresence = require('../presence/use-presence.cjs');
const presence_props = require('../presence/presence.props.cjs');
const datePickerContext = require('./date-picker-context.cjs');
const datePicker_props = require('./date-picker.props.cjs');
const useDatePicker = require('./use-date-picker.cjs');

const DatePickerRoot = /* @__PURE__ */ vue.defineComponent((props, {
  slots,
  attrs,
  emit
}) => {
  const api = useDatePicker.useDatePicker(props, emit);
  const isOpen = vue.computed(() => api.value.isOpen);
  const presenceProps = vue.computed(() => ({
    present: props.present || isOpen.value,
    lazyMount: props.lazyMount,
    unmountOnExit: props.unmountOnExit
  }));
  const presenceApi = usePresence.usePresence(presenceProps, emit);
  datePickerContext.DatePickerProvider(api);
  presenceContext.PresenceProvider(presenceApi);
  return () => vue.createVNode(factory.ark.div, vue.mergeProps(api.value.rootProps, attrs), {
    default: () => [slots.default?.(api.value)]
  });
}, {
  name: 'DatePickerRoot',
  props: {
    ...datePicker_props.props,
    ...presence_props.props
  },
  emits: [...datePicker_props.emits, ...presence_props.emits]
});

exports.DatePickerRoot = DatePickerRoot;
