// @ark-ui/vue/date-picker/date-picker.props.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const utils = require('../utils.cjs');

const props = {
  closeOnSelect: {
    type: Boolean,
    default: true
  },
  dir: {
    type: String
  },
  disabled: {
    type: Boolean,
    default: void 0
  },
  fixedWeeks: {
    type: Boolean
  },
  focusedValue: {
    type: Object
  },
  getRootNode: {
    type: Function
  },
  id: {
    type: String
  },
  ids: {
    type: Object
  },
  locale: {
    type: String
  },
  max: {
    type: Object
  },
  min: {
    type: Object
  },
  modal: {
    type: Boolean
  },
  modelValue: {
    type: Array
  },
  name: {
    type: String
  },
  numOfMonths: {
    type: Number
  },
  open: {
    type: Boolean
  },
  positioning: {
    type: Object
  },
  readOnly: {
    type: Boolean
  },
  selectionMode: {
    type: String
  },
  startOfWeek: {
    type: Number
  },
  timeZone: {
    type: String
  },
  view: {
    type: String,
    default: "day"
  }
};
const emits = utils.declareEmits([
  "focus-change",
  "value-change",
  "view-change",
  "open-change",
  "update:modelValue"
]);

exports.emits = emits;
exports.props = props;
