// @ark-ui/vue/date-picker/date-picker-table.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const vue = require('vue');
const factory = require('../factory.cjs');
const utils = require('../utils.cjs');
const datePickerContext = require('./date-picker-context.cjs');
const datePickerTableContext = require('./date-picker-table-context.cjs');
const datePickerViewContext = require('./date-picker-view-context.cjs');

const DatePickerTable = /* @__PURE__ */ vue.defineComponent((props, {
  slots,
  attrs
}) => {
  const api = datePickerContext.useDatePickerContext();
  const view = datePickerViewContext.useDatePickerViewContext();
  datePickerTableContext.DatePickerTableProvider(vue.reactive({
    ...props,
    id: utils.useId().value,
    ...view
  }));
  return () => vue.createVNode(factory.ark.table, vue.mergeProps(api.value.getTableProps(props), attrs), {
    default: () => [slots.default?.(api.value)]
  });
}, {
  name: 'DatePickerTable',
  props: {
    columns: {
      type: Number
    }
  }
});

exports.DatePickerTable = DatePickerTable;
