// @ark-ui/vue/date-picker/use-date-picker.mjs
import * as datePicker from '@zag-js/date-picker';
import { useMachine, normalizeProps } from '@zag-js/vue';
import { computed } from 'vue';
import { useEnvironmentContext } from '../environment/environment-context.mjs';
import { useId } from '../utils.mjs';

const useDatePicker = (props, emit) => {
  const getRootNode = useEnvironmentContext();
  const context = computed(() => {
    const { modelValue, focusedValue, ...rest } = props;
    return {
      ...rest,
      value: modelValue,
      focusedValue
    };
  });
  const [state, send] = useMachine(
    datePicker.machine({
      ...context.value,
      id: context.value.id ?? useId().value,
      getRootNode,
      onValueChange: (details) => {
        emit("value-change", details);
        emit("update:modelValue", details.value);
      },
      onFocusChange: (details) => {
        emit("focus-change", details);
      },
      onViewChange: (details) => {
        emit("view-change", details);
      },
      onOpenChange: (details) => {
        emit("open-change", details);
      }
    }),
    { context }
  );
  return computed(() => datePicker.connect(state.value, send, normalizeProps));
};

export { useDatePicker };
