// @ark-ui/vue/dialog/dialog.props.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const utils = require('../utils.cjs');

const props = {
  "aria-label": {
    type: String
  },
  closeOnEscapeKeyDown: {
    type: Boolean,
    default: void 0
  },
  closeOnInteractOutside: {
    type: Boolean,
    default: void 0
  },
  dir: {
    type: String
  },
  finalFocusEl: {
    type: [Function, Object]
  },
  getRootNode: {
    type: Function
  },
  id: {
    type: String
  },
  ids: {
    type: Object
  },
  initialFocusEl: {
    type: [Function, Object]
  },
  modal: {
    type: Boolean,
    default: void 0
  },
  open: {
    type: Boolean,
    default: void 0
  },
  preventScroll: {
    type: Boolean,
    default: void 0
  },
  restoreFocus: {
    type: Boolean,
    default: void 0
  },
  role: {
    type: String
  },
  trapFocus: {
    type: Boolean,
    default: void 0
  },
  defaultOpen: {
    type: Boolean
  }
};
const emits = utils.declareEmits([
  "escape-key-down",
  "open-change",
  "focus-outside",
  "interact-outside",
  "pointer-down-outside",
  "update:open"
]);

exports.emits = emits;
exports.props = props;
