// @ark-ui/vue/dialog/dialog-root.mjs
import { defineComponent, computed } from 'vue';
import { PresenceProvider } from '../presence/presence-context.mjs';
import { usePresence } from '../presence/use-presence.mjs';
import { props as props$1, emits as emits$1 } from '../presence/presence.props.mjs';
import { DialogProvider } from './dialog-context.mjs';
import { props, emits } from './dialog.props.mjs';
import { useDialog } from './use-dialog.mjs';

const DialogRoot = /* @__PURE__ */ defineComponent((props, {
  slots,
  emit
}) => {
  const api = useDialog(props, emit);
  const isOpen = computed(() => api.value.isOpen);
  const presenceProps = computed(() => ({
    present: props.present || isOpen.value,
    lazyMount: props.lazyMount,
    unmountOnExit: props.unmountOnExit
  }));
  const presenceApi = usePresence(presenceProps, emit);
  DialogProvider(api);
  PresenceProvider(presenceApi);
  return () => slots.default?.(api.value);
}, {
  name: 'DialogRoot',
  props: {
    ...props,
    ...props$1
  },
  emits: [...emits, ...emits$1]
});

export { DialogRoot };
