// @ark-ui/vue/dialog/dialog-root.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const vue = require('vue');
const presenceContext = require('../presence/presence-context.cjs');
const usePresence = require('../presence/use-presence.cjs');
const presence_props = require('../presence/presence.props.cjs');
const dialogContext = require('./dialog-context.cjs');
const dialog_props = require('./dialog.props.cjs');
const useDialog = require('./use-dialog.cjs');

const DialogRoot = /* @__PURE__ */ vue.defineComponent((props, {
  slots,
  emit
}) => {
  const api = useDialog.useDialog(props, emit);
  const isOpen = vue.computed(() => api.value.isOpen);
  const presenceProps = vue.computed(() => ({
    present: props.present || isOpen.value,
    lazyMount: props.lazyMount,
    unmountOnExit: props.unmountOnExit
  }));
  const presenceApi = usePresence.usePresence(presenceProps, emit);
  dialogContext.DialogProvider(api);
  presenceContext.PresenceProvider(presenceApi);
  return () => slots.default?.(api.value);
}, {
  name: 'DialogRoot',
  props: {
    ...dialog_props.props,
    ...presence_props.props
  },
  emits: [...dialog_props.emits, ...presence_props.emits]
});

exports.DialogRoot = DialogRoot;
