// @ark-ui/vue/segment-group/segment-group-item.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const vue = require('vue');
const anatomy = require('@ark-ui/anatomy');
const factory = require('../factory.cjs');
const segmentGroupContext = require('./segment-group-context.cjs');
const segmentGroupItemContext = require('./segment-group-item-context.cjs');

const SegmentGroupItem = /* @__PURE__ */ vue.defineComponent((props, {
  slots,
  attrs
}) => {
  const api = segmentGroupContext.useSegmentGroupContext();
  segmentGroupItemContext.SegmentGroupItemProvider(vue.computed(() => props));
  return () => vue.createVNode(factory.ark.label, vue.mergeProps(api.value.getItemProps(props), anatomy.segmentGroupAnatomy.build().item.attrs, attrs), {
    default: () => slots.default?.(api.value.getItemState(props))
  });
}, {
  name: 'SegmentGroupItem',
  props: {
    value: {
      type: String,
      required: true
    },
    disabled: {
      type: Boolean,
      default: undefined
    },
    invalid: {
      type: Boolean,
      default: undefined
    }
  }
});

exports.SegmentGroupItem = SegmentGroupItem;
