// @ark-ui/vue/factory.mjs
import { defineComponent, h } from 'vue';
import { Dynamic } from './dynamic.mjs';

const withAsChild = (component) => {
  const Polimoprhic = defineComponent({
    name: "Polimoprhic",
    inheritAttrs: false,
    props: {
      asChild: {
        type: Boolean,
        default: false
      }
    },
    setup(props, { attrs, slots }) {
      if (!props.asChild)
        return () => h(component, { ...attrs }, slots.default?.());
      return () => h(Dynamic, attrs, { default: slots.default });
    }
  });
  return Polimoprhic;
};
function jsxFactory() {
  const cache = /* @__PURE__ */ new Map();
  const factory = new Proxy(withAsChild, {
    apply(target, thisArg, argArray) {
      return withAsChild(argArray[0]);
    },
    get(_, element) {
      const asElement = element;
      if (!cache.has(asElement)) {
        cache.set(asElement, withAsChild(asElement));
      }
      return cache.get(asElement);
    }
  });
  return factory;
}
const ark = jsxFactory();

export { ark, jsxFactory };
