// @ark-ui/vue/factory.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const vue = require('vue');
const dynamic = require('./dynamic.cjs');

const withAsChild = (component) => {
  const Polimoprhic = vue.defineComponent({
    name: "Polimoprhic",
    inheritAttrs: false,
    props: {
      asChild: {
        type: Boolean,
        default: false
      }
    },
    setup(props, { attrs, slots }) {
      if (!props.asChild)
        return () => vue.h(component, { ...attrs }, slots.default?.());
      return () => vue.h(dynamic.Dynamic, attrs, { default: slots.default });
    }
  });
  return Polimoprhic;
};
function jsxFactory() {
  const cache = /* @__PURE__ */ new Map();
  const factory = new Proxy(withAsChild, {
    apply(target, thisArg, argArray) {
      return withAsChild(argArray[0]);
    },
    get(_, element) {
      const asElement = element;
      if (!cache.has(asElement)) {
        cache.set(asElement, withAsChild(asElement));
      }
      return cache.get(asElement);
    }
  });
  return factory;
}
const ark = jsxFactory();

exports.ark = ark;
exports.jsxFactory = jsxFactory;
